### DISCUSSION TITLE
[Common Question] What to do if my files failed to embed?

### DISCUSSION THREAD

**Discussion by pinuke** [brianpetro/obsidian-smart-connections#66] [Q&A] [answered]
# Common Question:

What do I do if I get an error stating that smart-connections failed to embed files like this:
`Smart Connections: Skipping previously failed file, use button in settings to retry`

![image](https://user-images.githubusercontent.com/12621101/226462281-abe09d0c-e58e-4b67-9cfc-828f4bf6c13b.png)

**Comment by pinuke** [comment, marked as answer]
Open up the plugin's settings and hit the "Retry failed files only" button

![image](https://user-images.githubusercontent.com/12621101/226462703-5907b736-3288-47fc-a37c-045636426466.png)

**Reply by pinuke** [reply]
Additionally, see [How to Check Your Rate Limit](https://github.com/brianpetro/obsidian-smart-connections/discussions/67)

**Comment by perryfr** [comment]
What to do when ALL your files fail to load? 

I've been using Smart Connections for a week now. No problems for the first few days, but then my vault kept on crashing. I found that turning off the Smart Connections plug-in resulted in my vault no longer crashing. Today I re-enabled the plug-in and asked ChatGPT for an answer to a question based on my notes, which it provided. Within minutes, the vault crashed again. 

I have tried several times to re-load the embeddings file, turned the plug-in on and off, closed and re-opened the vault, closed and re-opened Obsidian, checked to see whether I had missed a Smart Connections update. Nothing seems to work (and I still have money in my OpenAI account, I should add). I keep on getting the following messages in the Developer console.

<img width="389" alt="CleanShot 2023-07-07 at 14 52 08@2x" src="https://github.com/brianpetro/obsidian-smart-connections/assets/96356433/1546d8a9-7573-4373-ae98-01109d505b7b">

Any thoughts about how I proceed? Do I need to re-start from scratch? Thanks for whatever help you can provide.

**Reply by brianpetro** [reply]
@perryfr maybe the embeddings.json file got deleted somehow

Looks like you'll need to click the "Create embeddings" button in the "Smart View" window. Looks like this https://github.com/brianpetro/obsidian-smart-connections/issues/244#issuecomment-1603458823

**Comment by perryfr** [comment]
Nevermind! I figured it out. 

In case anyone else needs the following fairly basic assistance when using the plugin on a Mac, here it is: I used Cmd+Shift+G in Mac Finder to expose the .obsidian folder, clicked on the plugins folder, and then found the smart-connections folder. 

What my message was before I figured things out: Thanks for the response, @brianpetro. Does the embeddings.json file (in the .smart-connections folder) get stored in Obsidian's root folder (.obsidian)? In the future, I plan on regularly manually saving my embeddings file, so I want to make sure I know where I would want to put a saved embeddings file if something like this happens again. Thanks again for the great plug-in and for your assistance.

**Reply by brianpetro** [reply]
The `.smart-connections/` folder is stored at the root of your vault folder, alongside the `.obsidian/` folder, but not within it.

Thanks for the follow-up,
Brian 🌴
